### PR TITLE
release/1.5.0: site updates part 1 (Orion, Hercules, Narwhal)

### PR DESCRIPTION
**Pull request by climbfuji**
### Summary

- Update `.github/workflows/macos-ci-aarch64.yaml` - write spack caches to different partition (via a symbolic link) to avoid disk space issues
- Update site configs for Orion, Narwhal, Hercules: build spack python, update directory structure (3rd party dependencies, source cache), fix duplicate package problems
- Update documentation for Orion, Narwhal, Hercules

### Testing

- CI
- @climbfuji's macOS (arm64)
- Narwhal, Orion, Hercules (see https://github.com/JCSDA/spack-stack/issues/765)

### Applications affected

All applications on these three systems using spack-stack-1.5.0

### Systems affected

Orion, Narwhal, Hercules

### Dependencies

- [x] waiting on https://github.com/JCSDA/spack/pull/322

### Issue(s) addressed

- fixes https://github.com/JCSDA/spack-stack/issues/769

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so.
- [x] These changes have been tested on the affected systems and applications.
- [x] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

File: configs/sites/hercules/packages.yaml [hunk 1]
```diff
@@ -20,13 +20,12 @@ packages:
...
-    buildable: False
-    externals:
-    - spec: python@3.9.14 #+bz2+crypt+ctypes+dbm+lzma~nis+pyexpat~pythoncmd+readline+sqlite3+ssl~tix~tkinter+uuid+zlib
-      prefix: /usr
 
 ### Modifications of common packages
+  # "Soft suggestion" in common/packages.yaml doesn't work, need 5.4.1 because of gcc version
+  py-pyyaml:
+    require:
+    - "@5.4.1"
```
Comment (AlexanderRichert-NOAA) [inline configs/sites/hercules/packages.yaml]: How critical is it for awscli to be the latest version? As I mentioned in my Slack message, I ran into the problem of conflicting awscli/py-pyyaml/py-botocore versions, and was able to resolve it by taking awscli and py-pyyaml back a version from the latest.

**Review by ulmononian — APPROVED** [review]
might take some getting used to the new python set up (and require changes to the WM for orion for 1.5.0), but looks good.

File: configs/sites/hercules/packages.yaml (hunk 1 above)
Comment (climbfuji) [reply configs/sites/hercules/packages.yaml]: Based on the slack conversation we had, shall we merge this as is or should I change py-pyyaml to 5.4.1 and set awscli accordingly (which version is it), and remove the extra settings for Hercules/Orion?

Comment (AlexanderRichert-NOAA) [reply configs/sites/hercules/packages.yaml]: If it works for all systems to set the versions in common/packages.yaml (py-pyyaml 5.4.1, awscli 1.27.84) then let's go ahead and do that.

Comment (climbfuji) [reply configs/sites/hercules/packages.yaml]: Done.

**Review by AlexanderRichert-NOAA — APPROVED** [review]
:)

**Pull request merged**